bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -1017,32 +1018,66 @@ class MastodonClient private constructor(
                 .execute()
         }
 
+        /**
+         * Tries to get an [Instance] from the instance API endpoint.
+         *
+         * @param httpClient the HTTP client to use for the request
+         *
+         * @return the [Instance] retrieved from the server
+         * @throws InstanceRetrievalException if Instance could not be parsed from the server response
+         */
+        private fun getInstance(httpClient: OkHttpClient): Instance {
+            executeInstanceRequest(httpClient).use { response: Response ->
+                if (response.isSuccessful) {
+                    val body: String = response.body.string()
+                    try {
+                        val instance: Instance = JSON_SERIALIZER.decodeFromString(body)
+                        return instance
+                    } catch (e: SerializationException) {
+                        throw InstanceRetrievalException(
+                            "Could not decode response from server into Instance: $body",
+                            e
+                        )
+                    } catch (e: IllegalArgumentException) {
+                        throw InstanceRetrievalException(
+                            "Response from server does not appear to be a valid Instance: $body",
+                            e
+                        )
+                    }
+                } else {
+                    throw InstanceRetrievalException(
+                        "Instance request failed with code ${response.code} and message ${response.message}"
```

Comment:
Minor issue: `response.message` may be blank, leading to the confusing statement "Instance request failed with code 429 and message".